iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/dispatcher/IcebergTableEventDispatcher.java

```diff
@@ -74,17 +75,18 @@ public IcebergTableEventDispatcher(
 
   @Override
   public LoadTableResponse createTable(
-      String catalogName, Namespace namespace, CreateTableRequest createTableRequest) {
+      IcebergRequestContext context, Namespace namespace, CreateTableRequest createTableRequest) {
     TableIdentifier tableIdentifier = TableIdentifier.of(namespace, createTableRequest.name());
     NameIdentifier nameIdentifier =
-        IcebergRestUtils.getGravitinoNameIdentifier(metalakeName, catalogName, tableIdentifier);
+        IcebergRestUtils.getGravitinoNameIdentifier(
+            metalakeName, context.getCatalogName(), tableIdentifier);
     eventBus.dispatchEvent(
         new IcebergCreateTablePreEvent(
             PrincipalUtils.getCurrentUserName(), nameIdentifier, createTableRequest));
```

Comment:
Would it be possible to pass the whole context object inside pre-listener and post listener?

Reply:
Could you describe the scenarios?

Reply:
Sounds reasonable, @jerryshao WDYT?

Reply:
@jerryshao Hi Jerry, I think your idea works for filter checking. What about the use case of headers logging along with table events? How would that look like? Thanks.

Reply:
Hi @puchengy how about we merge this PR first, and discuss about your case in a separate issue?

Reply:
@jerryshao SG, thank you.